### PR TITLE
Add JFR ClassLoadingStatistics support

### PR DESCRIPTION
**Pull request by thallium**

### CODE REVIEW

**Comment by thallium** [comment]
@tajila FYI

File: runtime/vm/jfr.cpp [hunk 1]
```diff
@@ -965,6 +969,29 @@ jfrThreadCPULoadCallback(J9VMThread *currentThread, IDATA handlerKey, void *user
...
+	J9JFRClassLoadingStatistics *jfrEvent = (J9JFRClassLoadingStatistics *)reserveBuffer(currentThread, sizeof(J9JFRClassLoadingStatistics));
+
+	if (NULL != jfrEvent) {
+		initializeEventFields(currentThread, (J9JFREvent *)jfrEvent, J9JFR_EVENT_TYPE_CLASS_LOADING_STATISTICS);
+
+		UDATA unloadedClassCount = 0;
+		vm->memoryManagerFunctions->j9gc_get_cumulative_class_unloading_stats(currentThread, NULL, &unloadedClassCount, NULL);
+		jfrEvent->unloadedClassCount = (I_64)unloadedClassCount;
+
+		pool_state clState = {0};
```
Comment (tajila) [inline runtime/vm/jfr.cpp:985]: you should use the `allClassLoadersStartDo` API instead which acquires the correct class mutex

Comment (tajila) [reply runtime/vm/jfr.cpp:985]: `allClassLoadersNextDo` and `allClassLoadersEndDo`

Comment (tajila) [reply runtime/vm/jfr.cpp:985]: also, you need to release VM access before starting the classloading walk so you are not holding on to VM access while waiting for a lock

Comment (thallium) [reply runtime/vm/jfr.cpp]: fixed

File: runtime/vm/JFRChunkWriter.hpp [hunk 2]
```diff
@@ -600,6 +604,31 @@ class VM_JFRChunkWriter {
...
+		U_8 *dataStart = _bufferWriter->getAndIncCursor(sizeof(U_32));
+
+		/* write event type */
+		_bufferWriter->writeLEB128(ClassLoadingStatisticsID);
+
+		/* write start time */
+		_bufferWriter->writeLEB128(entry->ticks);
+
+		/* write user loaded class count */
+		_bufferWriter->writeFloat(entry->loadedClassCount);
```
Comment (tajila) [inline runtime/vm/JFRChunkWriter.hpp:623]: should be `writeLEB128`

File: runtime/vm/JFRChunkWriter.hpp [hunk 3]
```diff
@@ -600,6 +604,31 @@ class VM_JFRChunkWriter {
 		_bufferWriter->writeLEB128PaddedU32(dataStart, _bufferWriter->getCursor() - dataStart);
 	}
 
+	static void
```
Comment (tajila) [inline runtime/vm/JFRChunkWriter.hpp:607]: move this to the cpp file

Comment (thallium) [reply runtime/vm/JFRChunkWriter.hpp:607]: Why do we want that? Did we decide to move all the write functions to the cpp file?

Comment (tajila) [reply runtime/vm/JFRChunkWriter.hpp:607]: Yes, we should move all the write to .cpp. I made a mistake with the initial set. For now just move the new method, you can move the rest in a separate PR.

**Comment by tajila** [comment]
jenkins test sanity xlinux jdk17

**Comment by tajila** [comment]
@thallium Please create a PR for 0.49

**Pull request merged**